StoreTestWPF.ViewModel/ViewModels/StoreViewModel.cs

```diff
@@ -0,0 +1,206 @@
+            }
+        }
+
+        public ICommand EditCommand
+        {
+            get
+            {
+                return this.editCommand ?? new RelayCommandWithSender<CakeViewModel>(EditExecuted, EditCanExecute);
+            }
+        }
+
+        public ICommand DeleteCommand
+        {
+            get
+            {
+                return this.deleteCommand ?? new RelayCommandWithSender<CakeViewModel>(DeleteExecuted, DeleteCanExecute);
+            }
+        }
+
+        public StoreViewModel(IViewService viewService)
+        {
+            this.Cakes = new ObservableCollection<CakeViewModel>();
+            this.dbContext = new CakeStoreDbContext();
+            this.viewService = viewService;
+        }
+
+        private void AddExecuted()
+        {
+            ModifyCakeViewModel modifyViewModel = new ModifyCakeViewModel(this.viewService);
+            modifyViewModel.ModifiedCake = new CakeViewModel(new Cake());
+            modifyViewModel.WindowName = "Add cake";
+
+            if (this.viewService.ShowWindow(modifyViewModel) == true)
+            {
+                this.SelectedCake = modifyViewModel.ModifiedCake;
+                this.dbContext.Cakes.Add(SelectedCake.Cake);
+                this.dbContext.SaveChanges();
+
+                this.Cakes.Add(SelectedCake);
+            }
+        }
+
+        private void EditExecuted(CakeViewModel cake)
+        {
+            ModifyCakeViewModel modifyViewModel = new ModifyCakeViewModel(this.viewService);
+            modifyViewModel.ModifiedCake = cake;
+            modifyViewModel.WindowName = "Edit cake";
+
+            if (this.viewService.ShowWindow(modifyViewModel) == true)
+            {
+                this.dbContext.SaveChanges();
+
+                this.SelectedCake = modifyViewModel.ModifiedCake;
+            }
+        }
+
+        private bool EditCanExecute(CakeViewModel cake)
+        {
+            return cake != null && this.Cakes.Contains(cake);
+        }
+
+        private void DeleteExecuted(CakeViewModel cake)
+        {
+            if (this.viewService.ShowConfirmationMessage("Do you want to delete this cake permanently?") == true)
+            {
+                this.dbContext.Cakes.Remove(cake.Cake);
+                this.dbContext.SaveChanges();
+
+                this.Cakes.Remove(cake);
+            }
+        }
+
+        private bool DeleteCanExecute(CakeViewModel cake)
+        {
+            return cake != null && this.Cakes.Contains(cake);
+        }
+
+        private void LoadExecuted()
+        {
+            this.dbContext.Cakes.Load();
+            this.Cakes = new ObservableCollection<CakeViewModel>(dbContext.Cakes.Local.Select(cake => new CakeViewModel(cake)));
+            OnPropertyChanged(nameof(this.Cakes));
+        }
+
+        private bool LoadCanExecute()
+        {
+            return !this.Cakes.Any();
+        }
+
+        //to fill database initially
+        private void Init()
```

Comment:
As I can see the method is not used anywhere and should be removed.

Reply:
But Database should be filled with default values in another way.

Reply:
https://www.learnentityframeworkcore.com/migrations/seeding

Reply:
Fixed